cmd/signature-v4_test.go

```diff
@@ -221,6 +227,63 @@ func TestDoesPresignedSignatureMatch(t *testing.T) {
 			region:   serverConfig.GetRegion(),
 			expected: ErrSignatureDoesNotMatch,
 		},
+		// (8) Should error if the request is not ready yet, ie X-Amz-Date is in the future.
+		{
+			queryParams: map[string]string{
+				"X-Amz-Algorithm":      signV4Algorithm,
+				"X-Amz-Date":           now.Add(1 * time.Hour).Format(iso8601Format),
+				"X-Amz-Expires":        "60",
+				"X-Amz-Signature":      "badsignature",
+				"X-Amz-SignedHeaders":  "host;x-amz-content-sha256;x-amz-date",
+				"X-Amz-Credential":     fmt.Sprintf(credentialTemplate, serverConfig.GetCredential().AccessKeyID, now.Format(yyyymmdd), serverConfig.GetRegion()),
+				"X-Amz-Content-Sha256": payload,
```

Comment:
"X-Amz-Content-Sha256" can be removed as it's not part of the query params list http://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html

It's not harmful to have it, just that the test case suggests that it is also required. you can put a comment if you want to have it.

(same with other test cases)